### ISSUE TITLE
Rule proposal: <prefer-import-type>

### ISSUE THREAD

**Issue by dislido** [typescript-eslint/typescript-eslint#6268]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

prefer use `import type` instead of `import` if possible

### Fail Cases

```typescript
// lib.ts
export interface A {
  foo: number;
}

export class B {}
export const A: A = {foo: 123};

// index.ts
import {A, B} from 'lib';

const b = global.bInstance as B;
const a: A = {foo: 456};
```

### Pass Cases

```typescript
// lib.ts
export interface A {
  foo: number;
}

export class B {}
export const A: A = {foo: 123};

// index1.ts
import type {A, B} from 'lib';

const b = global.bInstance as B;
const a: A = {foo: 456};

// index2.ts
import {A, B} from 'lib'

console.log(A.foo);
const b = new B();
```

**Comment by Josh-Cena** [comment]
This is just [`consistent-type-imports`](https://typescript-eslint.io/rules/consistent-type-imports)?

**Comment by dislido** [comment]
oh I missed it, sorry
